### PR TITLE
feat: Add macOS target project

### LINKED ISSUE #256: Add support for macOS
**Problem Statement**

Add macOS target project

**Evidence or User Insights**

Allow macOS user to run this app natively.

**Proposal**

**Goals**

**Non-Goals**

**Low-Fidelity Concept**

### PR DESCRIPTION
**Pull request by ajpinedam**
## Fixes #.
closes #256 

### Description of the changes:
- Add macOS project
- Add some updates to make the app run
- Update Uno.UI to the latest 

### How changes were validated:

- Manual testing
-
-

### CODE REVIEW

File: .vsts-ci.macOS.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  pool:
+    name: Default
+    demands:
+    - Agent.OS -equals Darwin
+    - Xamarin.macOS -equals Stable
+
+  variables:
+    NUGET_PACKAGES: $(build.sourcesdirectory)/.nuget
+    VersionCodeOffset: 400559 # last value for the APK release
```
Comment (jeromelaban) [inline .vsts-ci.macOS.yml:9]: You can remove this

File: .vsts-ci.macOS.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
+jobs:
+- job: macOS
+
+  pool:
+    name: Default
+    demands:
```
Comment (jeromelaban) [inline .vsts-ci.macOS.yml]: use `vmImage: 'macOS-10.15'` instead

File: .vsts-ci.macOS.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    inputs:
+      solution: src/Calculator.sln
+      branchToMerge: uno
+      summaryFile: '$(Build.ArtifactStagingDirectory)/Canary.md'
+      resultFile: '$(Build.ArtifactStagingDirectory)/result.json'
+      nugetUpdaterVersion: 2.0.2
+      packageAuthor: nventive
+  
+  #- bash: /bin/bash -c "sudo $AGENT_HOMEDIRECTORY/scripts/select-xamarin-sdk.sh 5_18_1"
+  #  displayName: Select Xamarin Version
```
Comment (jeromelaban) [inline .vsts-ci.macOS.yml]: You can remove

File: .vsts-ci.macOS.yml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  
+  - bash: |
+      shopt -s nullglob
+      function join_by { local IFS="$1"; shift; echo "$*"; }
+      lib_path=$(join_by ';' $(Agent.WorkFolder)/_tasks/GitVersion*/4.0.*/lib/osx)
+      echo LD_LIBRARY_PATH: $lib_path
+      echo "##vso[task.setvariable variable=LD_LIBRARY_PATH]$lib_path"
+    displayName: Update LD_LIBRARY_PATH for GitVersion
```
Comment (jeromelaban) [inline .vsts-ci.macOS.yml]: This is not useful anymore

File: .vsts-ci.macOS.yml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    displayName: 'Install an Apple certificate'
+    inputs:
+      certSecureFile: 'nventive-appstore-prod-20190521.p12'
+      certPwd: '$(iOSSigning.CertificatePwd)'
+      signingIdentity: '$(iOSSigning.CertificateIdentity)'
+
+  - task: InstallAppleProvisioningProfile@1
+    displayName: 'Install an Apple provisioning profile'
+    inputs:
+      provProfileSecureFile: 'Uno_Calculator.mobileprovision'
```
Comment (jeromelaban) [inline .vsts-ci.macOS.yml:56]: Not sure we can reuse the same as iOS. Let's see if it works :)

File: .vsts-ci.macOS.yml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+  - task: NuGetToolInstaller@0
+    inputs:
+      versionSpec: 4.9.1
+      checkLatest: false
+
+  - task: NuGetCommand@2
+    inputs:
+      command: 'restore'
+      restoreSolution: '$(build.sourcesdirectory)/src/Calculator.macOS/Calculator.macOS.csproj'
```
Comment (jeromelaban) [inline .vsts-ci.macOS.yml:66]: You can remove this (the `/r` in the msbuild command below does this already)

File: .vsts-ci.macOS.yml [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      logProjectEvents: true
+      createLogFile: true
+
+  - task: MSBuild@1
+    inputs:
+      solution: '$(build.sourcesdirectory)/src/Calculator.macOS/Calculator.macOS.csproj'
+      msbuildLocationMethod: version
+      msbuildVersion: latest
+      msbuildArchitecture: x86
+      msbuildArguments: /r /p:Configuration=Release /p:Platform=iPhone /bl:$(build.artifactstagingdirectory)/build.binlog  "/p:InformationalVersion=$GITVERSION_INFORMATIONALVERSION"
```
Comment (jeromelaban) [inline .vsts-ci.macOS.yml]: Platform must be `iPhoneSimulator` for macOS

File: .vsts-ci.macOS.yml [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      clean: false
+      maximumCpuCount: true
+      restoreNugetPackages: false
+      logProjectEvents: true
+      createLogFile: true
+
+  - task: CopyFiles@2
+    displayName: 'Publish macOS Binaries'
+    inputs:
+      SourceFolder: $(build.sourcesdirectory)/src/Calculator.macOS/bin/iPhone/Release
```
Comment (jeromelaban) [inline .vsts-ci.macOS.yml]: same here `iPhoneSimulator`

File: src/Calculator.macOS/MenuHelper.cs [hunk 9]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using AppKit;
+
+namespace calculator.macOS
+{
+	public static class MenuHelper
+	{
+		internal static NSMenu GetMenu()
+		{
+			var menubar = new NSMenu();
```
Comment (jeromelaban) [inline src/Calculator.macOS/MenuHelper.cs:10]: You should be able to use the MenuBar control, with a style:

https://github.com/unoplatform/uno/blob/b7ccfcae1e4a76044b86e572802049869d9e746e/src/Uno.UI/UI/Xaml/Controls/MenuBar/MenuBar.xaml#L31

File: src/Calculator.macOS/Properties/AssemblyInfo.cs [hunk 10]
```diff
@@ -0,0 +1,36 @@
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("$projectname$")]
```
Comment (jeromelaban) [inline src/Calculator.macOS/Properties/AssemblyInfo.cs:8]: Looks like it did not get changed, nor in the rest of the file... :) A bug for the templates.

File: src/Calculator.macOS/AppDelegate.cs [hunk 11]
```diff
@@ -0,0 +1,7 @@
+using AppKit;
+using Foundation;
+
+namespace calculator.macOS
```
Comment (davidjohnoliver) [inline src/Calculator.macOS/AppDelegate.cs]: Should be capitalized - and is this empty file actually needed?

File: src/Calculator.macOS/Main.cs [hunk 12]
```diff
@@ -0,0 +1,17 @@
+using AppKit;
+using CalculatorApp;
+
+namespace calculator.macOS
```
Comment (davidjohnoliver) [inline src/Calculator.macOS/Main.cs]:
```suggestion
namespace WindowsCalculator.MacOS
```

File: src/Calculator.macOS/MenuHelper.cs [hunk 13]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using AppKit;
+
+namespace calculator.macOS
```
Comment (davidjohnoliver) [inline src/Calculator.macOS/MenuHelper.cs]:
```suggestion
namespace WindowsCalculator.MacOS
```

**Pull request merged**